src/librustc_mir/transform/const_prop.rs

```diff
@@ -334,6 +334,13 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
         )
         .expect("failed to push initial stack frame");
 
+        let mut only_propagate_inside_block_locals = BitSet::new_empty(can_const_prop.len());
+        for (l, mode) in can_const_prop.iter_enumerated() {
```

Comment:
Ohh! I don't think I realized that doing this construction here was legal. This is nice :)

So, just to check: this code is executed once per block, right? @wesleywiser

Reply:
Otherwise, I don't think this change is 100% sound. 
Around line 1082, we might be removing consts from the const pool that are not from the current block, yet are restricted to propagate only into their own block. If that makes sense?

Reply:
Well, it passed the tests, so either it's sound, or it threaded the needle through our tests.
Still I think that the failure mode of this expansion to const-prop (if it is buggy) is to propagate less, and not propagate more constants nor the wrong constants. So even if it's buggy, it shouldn't be actually unsound.
To catch those cases we'll need more tests. I hope I can work on that after I finish getting through bureaucracy with uni >.<